flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/test/java/com/github/shyiko/mysql/binlog/event/deserialization/TableMapEventMetadataDeserializerTest.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.shyiko.mysql.binlog.event.deserialization;
+
+import com.github.shyiko.mysql.binlog.event.TableMapEventMetadata;
+import com.github.shyiko.mysql.binlog.io.ByteArrayInputStream;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Tests for the copied class {@link TableMapEventMetadataDeserializer}. */
+public class TableMapEventMetadataDeserializerTest {
+    @Test
+    public void testDeserialize() throws IOException {
+        TableMapEventMetadataDeserializer metadataDeserializer =
+                new TableMapEventMetadataDeserializer();
+
+        byte[] data = {1, 2, 64, 0, 2, 3, -4, -1, 0};
+
+        TableMapEventMetadata tableMapEventMetadata =
+                metadataDeserializer.deserialize(new ByteArrayInputStream(data), 10, 8);
+
+        assertThat(tableMapEventMetadata.getSignedness().get(1));
```

Comment:
The result is not checked here.

Reply:
Could you please add some ITCase to reproduce the problem?

Reply:
@ruanhang1993 thank you for your review, I've modified the description and unit tests, the bytes used in my test case can be reproduced before modifying the code. I also mentioned the conditions for reproduction in https://github.com/ververica/flink-cdc-connectors/issues/2784